ext/stringio/stringio.c

```diff
@@ -1208,20 +1208,23 @@ strio_getline(struct getline_arg *arg, struct StringIO *ptr)
 	    }
 	}
 	s = p;
+        const char *paragraph_end = NULL;
```

Comment:
Still supporting older ruby using C89.
Move this definition to the top of the block.